File: 21/solution.py
def getFactors(target):
	factors = []
	for i in xrange(1, int(target ** 0.5)):
		if target % i == 0:
			factors.append(i)
			factors.append(target / i)
	if target in factors:
		factors.remove(target)
	return factors

amicable_numbers = []
for i in range(1, 10000):
	if i in amicable_numbers:
		continue
	total = sum(getFactors(i))
	if sum(getFactors(total)) == i and total != i:
		amicable_numbers.append(i)
		amicable_numbers.append(total)

print sum(amicable_numbers)

File: 21/solution.js
function getFactors(target) {
	var factors = [1];
	for (var i = 2; i < Math.sqrt(target); i += 1) {
		if (target % i === 0) {
			factors.push(i);
			factors.push(target / i);
		}
	}
	return factors;
}

var amicable_numbers = [];
for (var candidate = 1; candidate < 10000; candidate += 1) {
	if (amicable_numbers.indexOf(candidate) !== -1) {
		continue;
	}
	var candidate_factors = getFactors(candidate);
	var total = 0
	for (var i = 0; i < candidate_factors.length; i += 1) {
		total += candidate_factors[i];
	}
	var total_factors = getFactors(total);
	var total_total = 0;
	for (var i = 0; i < total_factors.length; i += 1) {
		total_total += total_factors[i];
	}
	if (total_total === candidate && total !== candidate) {
		amicable_numbers.push(total);
		amicable_numbers.push(candidate);
	}
}

var sum = 0
for (var i = 0; i < amicable_numbers.length; i += 1) {
	sum += amicable_numbers[i];
}

console.log(sum);